templates/skeleton/app/components/Header.tsx

```diff
@@ -159,15 +163,19 @@ function CartToggle({cart}: Pick<HeaderProps, 'cart'>) {
   return (
     <Suspense fallback={<CartBadge count={null} />}>
       <Await resolve={cart}>
-        {(cart) => {
-          if (!cart) return <CartBadge count={0} />;
-          return <CartBadge count={cart.totalQuantity || 0} />;
-        }}
+        <CartBanner />
       </Await>
     </Suspense>
   );
 }
 
+function CartBanner() {
+  const cart = useAsyncValue();
+  const optimisticCart = useOptimisticCart(cart);
```

Comment:
Nit: just for consistency can you [follow the same naming pattern as here](https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/components/CartMain.tsx#L19-L22)?

Something like:

```
const originalCart = useAsyncValue();
const cart = useOptimisticCart(originalCart);
...
```